### DISCUSSION TITLE
How to output derivatives in csv or exodus?

### DISCUSSION THREAD

**Discussion by aiskhak** [idaholab/moose#20004] [Q&A General] [answered]
I have a FV Navier-Stokes application. I need to output velocity derivatives at each spatial location in csv or exodus file. How do I do that?

**Comment by GiudGiud** [comment, marked as answer]
Hello

If you update your version of MOOSE you should be able to use a GenericFunctorGradientMaterial to get the derivatives of the velocity into a vector functor material property. 

Then you can use the appropriate aux kernel (GenericFunctorVectorAux -> FunctorVectorElementalAux I think) to output it to an aux variable

Guillaume
